options/cf_options.cc

```diff
@@ -506,204 +549,233 @@ std::unordered_map<std::string, OptionTypeInfo>
           }}},
         {"table_factory",
          {offset_of(&ColumnFamilyOptions::table_factory),
-          OptionType::kTableFactory, OptionVerificationType::kByName,
-          OptionTypeFlags::kCompareLoose, 0}},
+          OptionType::kConfigurable, OptionVerificationType::kByName,
+          (OptionTypeFlags::kShared | OptionTypeFlags::kCompareLoose |
+           OptionTypeFlags::kDontPrepare),
```

Comment:
what's the reason that `table_factory` should not call `PrepareOptions`?

Reply:
It is because of how the table factory representation is stored in the Options file.  Inside of the options file, the "Column Family" will contain something like "table_factory = BlockBased" with just the class of the TableFactory being used.  A later section of the Options file will have the configuration for the TableFactory for that Column Family Option".  Because the table factory ends up being configured (and prepared) later, it should not be prepared as part of the column family but as part of the setup of the table factory (at least when configured via the Options file).

OTOH, the "block_based_table_factory" and "plain_table_factory" options are used for testing and setup (not persisting the options of the table factory).  Therefore they can be configured earlier.